### PR TITLE
Improve installation script for code deploy

### PR DESCRIPTION
**Pull request by matjazv**
### What was the problem?

This PR resolves LISK-912.

### How was it solved?

- [x] Execute `installDependencies.sh` script inside `AfterInstall` hook.
- [x] Execute scripts inside `appspec.yml` as root.
- [x] Use `--frozen-lockfile` for `yarn install`
- [x] Read and use node version from `.nvmrc` file

### How was it tested?

`installDependencies.sh` script was locally executed

### CODE REVIEW

File: scripts/installDependencies.sh [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-# Script to install Node.js using NVM on Ubuntu without sudo
+# Script to install Node.js using NVM on Ubuntu
+
+# Read node version from .nvmrc file
+node_version=$(cat .nvmrc | tr -d "\n")
 
-node_version=v20
 # Function to install NVM
 install_nvm() {
     curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
```
Comment (sameersubudhi) [inline scripts/installDependencies.sh:11]: Optional, but we can already bump to the latest available nvm. We can make this to be dynamic in the future.
```suggestion
    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.0/install.sh | bash
```

File: scripts/installDependencies.sh [hunk 2]
```diff
@@ -19,34 +21,25 @@ install_node_version() {
...
-    else
-        echo "Node version $node_version is not installed. Installing..."
-        nvm install $node_version
-        nvm use $node_version
+        source $nvm_path
     fi
+    
+    echo "Installing Node version $node_version..."
+    nvm install $node_version
+    nvm use $node_version
```
Comment (sameersubudhi) [inline scripts/installDependencies.sh:33]: Optional, running `nvm use` after `nvm install` is redundant. `nvm install` already ensures that the current installed version is used.
```suggestion
    nvm install $node_version
```

**Review by sameersubudhi — APPROVED** [review]
LGTM! 👍
Have left a couple of optional points to address.

File: scripts/installDependencies.sh (hunk 1 above)
Comment (matjazv) [reply scripts/installDependencies.sh]: Done.

File: scripts/installDependencies.sh (hunk 2 above)
Comment (matjazv) [reply scripts/installDependencies.sh]: Done.

File: scripts/installDependencies.sh [hunk 3]
```diff
@@ -19,34 +21,24 @@ install_node_version() {
...
+    echo "Installing Node version $node_version..."
+    nvm install $node_version
 }
 
 install_node_version
 
 echo "Node.js version $node_version has been installed."
 
-sudo apt install npm
-echo "***Installed npm***"
```
Comment (ishantiw) [inline scripts/installDependencies.sh:46]: I think you need to install `npm` for further steps

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: scripts/installDependencies.sh (hunk 3 above)
Comment (sameersubudhi) [reply scripts/installDependencies.sh:46]: IIRC, npm is automatically installed with nvm. Is it not? Also, installing NodeJS should also install npm, by default

**Review by ishantiw — APPROVED** [review]
👏

**Pull request merged**